client/my-sites/plugins/plugin-details.jsx

```diff
@@ -221,8 +221,21 @@ function PluginDetails( props ) {
 			},
 		];
 
+		const pluginsSearchId = 'plugins-search';
+		if ( breadcrumbs.length > 1 && breadcrumbs[ 1 ].id === pluginsSearchId ) {
+			const searchHref = breadcrumbs[ 1 ].href;
+			const searchQuery = searchHref.substring( searchHref.lastIndexOf( '?' ) );
+			items.push( {
+				helpBubble: null,
+				label: 'Search Results',
+				href: `/plugins/${ selectedSite?.slug || '' }${ searchQuery || '' }`,
+				id: pluginsSearchId,
+			} );
+		}
```

Comment:
Reaching into the state we're about to mutate feels wrong for some reason. Not quite sure I can explain it.

[Replacing the append calls with updates](https://github.com/Automattic/wp-calypso/pull/66794) might have been the wrong way to fix this. Should we revert that or work on a [better solution](pdh6GB-1QJ-p2#comment-2778)?

Reply:
> Reaching into the state we're about to mutate feels wrong for some reason. Not quite sure I can explain it.

Yeah, I know what you mean.

Maybe a revert of #66794 and start from the top?

Reply:
Revert up here https://github.com/Automattic/wp-calypso/pull/67071